test/Datadog.Trace.Tests/CorrelationIdentifierTests.cs

```diff
@@ -9,24 +11,92 @@ namespace Datadog.Trace.Tests
-            Assert.Equal<ulong>(parentSpan.TraceId, CorrelationIdentifier.TraceId);
-            parentScope.Close();
+        [Fact]
+        public void TraceIdSpanId_ZeroOutsideActiveSpan()
+        {
+            using (var parentScope = Tracer.Instance.StartActive("parent"))
+            using (var childScope = Tracer.Instance.StartActive("child"))
+            {
+                // Do nothing
+            }
 
             Assert.Equal<ulong>(0, CorrelationIdentifier.SpanId);
             Assert.Equal<ulong>(0, CorrelationIdentifier.TraceId);
         }
+
+        [Fact]
+        public void ServiceIdentifiers_MatchTracerInstanceSettings()
+        {
+            const string service = "unit-test";
+            const string version = "1.0.0";
+            const string env = "staging";
+
+            var settings = new TracerSettings()
+            {
+                ServiceName = service,
+                ServiceVersion = version,
+                Environment = env
+            };
+            var tracer = new Tracer(settings);
+            Tracer.Instance = tracer;
+
+            using (var parentScope = Tracer.Instance.StartActive("parent"))
+            using (var childScope = Tracer.Instance.StartActive("child"))
+            {
+                Assert.Equal(service, CorrelationIdentifier.Service);
+                Assert.Equal(version, CorrelationIdentifier.Version);
+                Assert.Equal(env, CorrelationIdentifier.Env);
+            }
+
+            Assert.Equal(service, CorrelationIdentifier.Service);
+            Assert.Equal(version, CorrelationIdentifier.Version);
+            Assert.Equal(env, CorrelationIdentifier.Env);
+        }
+
+        [Fact]
+        public void VersionAndEnv_EmptyStringIfUnset()
+        {
+            var settings = new TracerSettings();
+            var tracer = new Tracer(settings);
+            Tracer.Instance = tracer;
+
+            using (var parentScope = Tracer.Instance.StartActive("parent"))
+            using (var childScope = Tracer.Instance.StartActive("child"))
+            {
+                Assert.Equal(string.Empty, CorrelationIdentifier.Version);
+                Assert.Equal(string.Empty, CorrelationIdentifier.Env);
+            }
+
+            Assert.Equal(string.Empty, CorrelationIdentifier.Version);
+            Assert.Equal(string.Empty, CorrelationIdentifier.Env);
+        }
+
+        [Fact]
+        public void Service_DefaultServiceNameIfUnset()
+        {
+            var settings = new TracerSettings();
+            var tracer = new Tracer(settings);
+            Tracer.Instance = tracer;
+
+            using (var parentScope = Tracer.Instance.StartActive("parent"))
+            using (var childScope = Tracer.Instance.StartActive("child"))
+            {
+                Assert.Contains(CorrelationIdentifier.Service, Tracer.Instance.DefaultServiceName);
+            }
+
+            Assert.Contains(CorrelationIdentifier.Service, Tracer.Instance.DefaultServiceName);
```

Comment:
Contains? Are we comparing substrings?

Reply:
Whoops, fixed. I started off thinking I'd have to validate that the name existed in the `IEnumerable<string> Datadog.Trace.TestHelpers.TestRunners.ValidNames` but I ended up not needing to do that. I've fixed this so it's checking an exact string match